app/components/zitting-text-document-container.js

```diff
@@ -0,0 +1,58 @@
+import Component from '@glimmer/component';
+import { task } from "ember-concurrency-decorators";
+import { action } from "@ember/object";
+
+export default class ZittingTextDocumentContainerComponent extends Component {
+  constructor(...args) {
+    super(...args);
+    this.getText.perform();
+  }
+
+  profile = 'none';
+  editor;
+  type = this.args.type;
+  text;
+
+  editorOptions = {
+    showToggleRdfaAnnotations: false,
+    showInsertButton: true,
+    showRdfa: false,
+    showRdfaHighlight: false,
+    showRdfaHover: false
+  };
+
+  @task
+  *getText() {
```

Comment:
If you keep the logic in the component, I would convert this to a getter instead.